OTAnalytics/application/use_cases/highlight_intersections.py

```diff
@@ -59,6 +63,176 @@ def get_ids(self) -> set[TrackId]:
         ).get_ids()
 
 
+class TracksIntersectingAllNonCuttingSections(TrackIdProvider):
+    """Returns track ids intersecting all sections which are not a cutting section.
+
+    Args:
+        get_all_sections (GetAllSections): the use case to get all sections.
+        tracks_intersecting_sections (TracksIntersectingSections): get track ids
+            intersecting sections.
+        get_section_by_id (GetSectionsById): use case to get sections by id.
+    """
+
+    def __init__(
+        self,
+        get_cutting_sections: GetCuttingSections,
+        get_all_sections: GetAllSections,
+        tracks_intersecting_sections: TracksIntersectingSections,
+        get_section_by_id: GetSectionsById,
+        intersection_repository: IntersectionRepository,
+    ) -> None:
+        self._get_cutting_sections = get_cutting_sections
+        self._get_all_sections = get_all_sections
+        self._tracks_intersecting_sections = tracks_intersecting_sections
+        self._get_section_by_id = get_section_by_id
+        self._intersection_repository = intersection_repository
+
+    def get_ids(self) -> set[TrackId]:
+        return TracksIntersectingGivenSections(
```

Comment:
Instead of `TracksIntersectingGivenSections` a new use case is necessary. Because `TracksIntersectingGivienSections` calculates line intersections. No track has a line intersection with a cutting section.
The new use case should be similar to `SimpleTracksIntersectingSections` in `OTAnalytics/plugin_intersect/simple_intersect.py`. Instead of using `intersecting_tracks` of `TrackDataset` in line 25, use `contained_by_sections`. This will check if detections are inside the area of the given sections.